IdentityCore/src/webview/embeddedWebview/challangeHandlers/ios/MSIDCertAuthHandler.m

```diff
@@ -136,30 +136,28 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
     NSDictionary *queryItemsDict = [NSDictionary msidDictionaryFromQueryItems:queryItems];
     NSString *redirectURI = queryItemsDict[MSID_OAUTH2_REDIRECT_URI];
 
-    if (s_redirectScheme)
+    NSMutableDictionary *newQueryItems = [NSMutableDictionary new];
+    newQueryItems[MSID_BROKER_IS_PERFORMING_CBA] = @"true";
+    
+    for (NSURLQueryItem *item in queryItems)
     {
-        NSMutableDictionary *newQueryItems = [NSMutableDictionary new];
-        NSString *redirectSchemePrefix = [NSString stringWithFormat:@"%@://", s_redirectScheme];
-        
-        newQueryItems[MSID_BROKER_IS_PERFORMING_CBA] = @"true";
-        for (NSURLQueryItem *item in queryItems)
+        if ([item.name isEqualToString:MSID_OAUTH2_REDIRECT_URI] && !s_useAuthSession && s_redirectScheme != nil)
         {
-            if ([item.name isEqualToString:MSID_OAUTH2_REDIRECT_URI]
-                && ![item.value.lowercaseString hasPrefix:redirectSchemePrefix.lowercaseString]
-                && !s_useAuthSession)
+            NSString *redirectSchemePrefix = [NSString stringWithFormat:@"%@://", s_redirectScheme];
+            if (![item.value.lowercaseString hasPrefix:redirectSchemePrefix.lowercaseString])
             {
                 newQueryItems[MSID_OAUTH2_REDIRECT_URI] = [s_redirectPrefix stringByAppendingString:item.value.msidURLEncode];
```

Comment:
Not really your change but why value need to be URL encode here, woud it be double encoding in line 157?

Reply:
it is not double encoded, it is set as a string in broker.

Reply:
Can you clarify why it can avoid double encode, sorry I don't get it yet. My test:
original string: "&"
Set into newQueryItems (line 149) => change value to "%26"
later, in line 157, it will be encoded again:
value change to "%2526" 
so requestURLComponents.percentEncodedQuery will have the value of key MSID_OAUTH2_REDIRECT_URI double encoded.

Reply:
From the code, I think the value of redirectURI (line 159) will not be double-encoded but value of requestURLComponents.percentEncodedQuery (line 157) will be double-encoded. I'm not sure the risk of that but just want to call-out the expected value here.